src/akkudoktoreos/core/pydantic.py

```diff
@@ -51,6 +52,71 @@ def deep_update(source_dict: dict[str, Any], update_dict: dict[str, Any]) -> dic
     return merged_dict
 
 
+def access_nested_value(
+    model: BaseModel, path: Union[str, Sequence[str]], setter: bool, value: Optional[Any] = None
+) -> Any:
+    """Get or set a nested model value based on the provided path.
+
+    Supports string paths (with '/' separators) or sequence paths (list/tuple).
+    Trims leading and trailing '/' from string paths.
+
+    Args:
+        model (BaseModel): The model object for partial assignment.
+        path (Union[str, Sequence[str]]): The path to the model key (e.g., "key1/key2/key3" or ["key1", "key2", "key3"]).
```

Comment:
What about ints in the Sequence to denote list elements?

Reply:
Good point, however it's not used atm and I didn't write any tests for it, so I would remove the Sequence support.